### ISSUE TITLE
File validation doesn't start on prod

### ISSUE THREAD

**Issue by prabh-t** [ebi-ait/hca-ebi-dev-team#269]
Marion:
My next issue is that even though I am transferring the files to the upload bucket, the ui isn't registering the files are there so it apparently isn't validating.
submission: https://ui.ingest.archive.data.humancellatlas.org/submissions/detail?uuid=52e21998-8500-46db-8f22-ea6b1eaf67c0&project=83f5188e-3bf7-4956-9544-cea4f8997756
ingest upload area:
```
s3://org-hca-data-archive-upload-prod/52e21998-8500-46db-8f22-ea6b1eaf67c0/
```
if you `aws s3 ls`  the above area, most of the files are there, but no files in the submission are transitioning through validating/valid states

**Comment by prabh-t** [comment]
Rodrey:
It might also have been a hiccup with the AWS infrastructure yesterday.

Prabhat:
..it seems the transfer (and notification) did happen without error but upload service failed to trigger the validation.
I doubt we're missing a step as it works on the other envs. It appears to be a setup/config issue on prod.

**Comment by prabh-t** [comment]
Rodrey:
It might also have been a hiccup with the AWS infrastructure yesterday.

Prabhat:
..it seems the transfer (and notification) did happen without error but upload service failed to trigger the validation.
I doubt we're missing a step as it works on the other envs. It appears to be a setup/config issue on prod.

**Comment by rdgoite** [comment]
Based on the findings from yesterday this problem was due to 2 core issues. 

* First, the HCA util needs to be modified to encode safe URLs, as file names with special characters such as # can lead to invalid API calls. 
* The other issue is that the data files were uploaded without `dcp-type` associated in their records. This could have been an issue specific to the datasets listed here, but currently, the `dcp-type` is required for the operations to flow smoothly.

Additionally, we need to update the file update listener in the core to display logs on its operation. The listener kept silent about some important details in its processing. It would have helped if the listener warned about receiving update messages with no `dcp-type` data attached to them.

**Comment by rdgoite** [comment]
At the time of writing, there are 2 options to work around this issue:

1. Reupload all the files using HCA CLI.
1. Update file records in the upload service with their proper dcp-type (which is "data") in this case.

Option 2 involves calling the Upload Service API:

1. Retrieve the file info from the upload service using `GET /v1/area/{upload_area_uuid}/{filename}`.
1. Update the file using exactly the same info through `PUT /v1/area/{upload_area_uuid}/{filename}`. This requires an API key header, and the Content-type header with ; data-type=data appended to it.
1. Once the update is done, notify the upload service using `PUT /v1/area/{upload_area_uuid}/{filename}`.

**Comment by prabh-t** [comment]
Thanks Rodrey. The hca-cli is adding the parameter part of the content-type at a different step in the process, the hca-util is doing so at the upload step. but since this was skipped, and no check was in place at the sync step, upload service is not having it right.

An update to the hca-util (v0.2.11) has been published that addresses the issues with the tool.

1. path params now encoded to handle potential unsafe character, in this case, in {filename} in `POST /v1/area/{upload_area_uuid}/{filename}` to notify upload service
2. added a check in sync command to ensure dcp-type is appended in the file content-type

PR for this fix is [here](https://github.com/ebi-ait/hca-util/pull/14).

**Comment by lauraclarke** [comment]
@prabh-t has this fix been deployed or at least approved for deployment?

**Comment by aaclan-ebi** [comment]
@prabh-t 's fix is in the hca-util's latest version `v0.2.12`

**Comment by aaclan-ebi** [comment]
I believe this was accomplished in the last sprint/s and has already been released to prod. Closing now.
